From d89c48638f8a91b9df06d00f41a683591c7f3469 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 7 Jun 2010 22:37:20 +0100
Subject: [PATCH 1/2] Staging: rtl8192s_usb: Fix device ID table

The rtl8192s_usb driver was claiming a number of device IDs which are
actually supported by the code in the rtl8192u_usb driver.  Remove
them from rtl8192s_usb, and add more device IDs as listed in the
vendor driver version 0003.0825.2009.
---
 drivers/staging/rtl8192su/r8192U_core.c |   28 +++++++++++++++++-----------
 1 files changed, 17 insertions(+), 11 deletions(-)

diff --git a/drivers/staging/rtl8192su/r8192U_core.c b/drivers/staging/rtl8192su/r8192U_core.c
index 03f1cf1..a0c28a7 100644
--- a/drivers/staging/rtl8192su/r8192U_core.c
+++ b/drivers/staging/rtl8192su/r8192U_core.c
@@ -114,26 +114,32 @@ u32 rt_global_debug_component = \
 static const struct usb_device_id rtl8192_usb_id_tbl[] = {
 	/* Realtek */
 	{USB_DEVICE(0x0bda, 0x8171)},
-	{USB_DEVICE(0x0bda, 0x8192)},
-	{USB_DEVICE(0x0bda, 0x8709)},
+	{USB_DEVICE(0x0bda, 0x8172)},	/* RTL8191S */
+	{USB_DEVICE(0x0bda, 0x8173)},
+	{USB_DEVICE(0x0bda, 0x8174)},
+	{USB_DEVICE(0x0bda, 0x8712)},
+	{USB_DEVICE(0x0bda, 0x8713)},
 	/* Corega */
-	{USB_DEVICE(0x07aa, 0x0043)},
+	{USB_DEVICE(0x07aa, 0x0047)},
 	/* Belkin */
-	{USB_DEVICE(0x050d, 0x805E)},
 	{USB_DEVICE(0x050d, 0x815F)}, /* Belkin F5D8053 v6 */
 	/* Sitecom */
-	{USB_DEVICE(0x0df6, 0x0031)},
+	{USB_DEVICE(0x0df6, 0x0045)},	/* WL-353 */
 	{USB_DEVICE(0x0df6, 0x004b)},	/* WL-349 */
 	/* EnGenius */
-	{USB_DEVICE(0x1740, 0x9201)},
+	{USB_DEVICE(0x1740, 0x9603)},
+	{USB_DEVICE(0x1740, 0x9605)},
 	/* Dlink */
-	{USB_DEVICE(0x2001, 0x3301)},
-	/* Zinwell */
-	{USB_DEVICE(0x5a57, 0x0290)},
+	{USB_DEVICE(0x07d1, 0x3303)},
+	{USB_DEVICE(0x07d1, 0x3302)},	/* DWA-130 rev C2 */
+	{USB_DEVICE(0x07d1, 0x3300)},	/* DWA-130 rev E */
 	/* Guillemot */
 	{USB_DEVICE(0x06f8, 0xe031)},
-	//92SU
-	{USB_DEVICE(0x0bda, 0x8172)},
+	/* Edimax */
+	{USB_DEVICE(0x7392, 0x7612)},
+	/* Hawking */
+	{USB_DEVICE(0x0e66, 0x0015)},
+	{USB_DEVICE(0x0e66, 0x0016)},
 	{}
 };
 
-- 
1.7.1

